adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -189,4 +160,46 @@ - (void)unityAdsAdLoaded:(nonnull NSString *)placementId {
   }
 }
 
+#pragma mark - UnityAdsShowDelegate Methods
+
+- (void)unityAdsShowClick:(NSString *)placementId {
+  // The Unity Ads SDK doesn't provide an event for leaving the application, so the adapter assumes
+  // that a click event indicates the user is leaving the application for a browser or deeplink, and
+  // notifies the Google Mobile Ads SDK accordingly.
+  [_adEventDelegate reportClick];
+}
+
+- (void)unityAdsShowComplete:(NSString *)placementId withFinishState:(UnityAdsShowCompletionState)state {
+  if (state == kUnityAdsFinishStateCompleted) {
```

Comment:
I would probably separate the if/else state into separate functions, to help with readability.

```objc
self.processIfCompleted(state, placementId)
self.reportIfError(state)
```
just thoughts.